### PR TITLE
Cache Python Packages in the CI pipeline

### PR DESCRIPTION
**Pull request by merwanehamadi**
### Changes

Now python packages are cached in the ci pipeline.
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes. 
- [x] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports autogpt tests --in-place
    ```

### CODE REVIEW

**Review by Pwuts — CHANGES_REQUESTED** [review]
This would save a considerable amount of time to run the CI, nice! That cache needs to be cleaned regularly though, similar to the `docker-ci` and `docker-cache-clean` workflows.

**Comment by merwanehamadi** [comment]
@Pwuts 
- removed useless exit 0
- change git diff command to spot change in current_score to make it more precise
- skipped a test that unfortunately we cannot maintain right now.

**Comment by BaseInfinity** [comment]
This looks good to me! Quick question:
1. What's the idea with including the date?

**Comment by merwanehamadi** [comment]
@BaseInfinity have the latest packages every day. otherwise, if our requirements.txt doesn't change for 2 weeks, our packages go stale.

File: tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py [hunk 1]
```diff
@@ -14,6 +14,9 @@
 CYCLE_COUNT = 3
 
 
+@pytest.mark.skip(
+    "This test uses the browse website functionality, which VCR doesn't support yet, this means it's slower, costs money and is less reliable. It will be unskipped once VCR support browse_website"
+)
```
Comment (Pwuts) [inline tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py]: Why should `browse_website` rely on VCR? Requests to websites don't cost us any money, and with the right choice of content, we can trust that the website is static over time.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -32,6 +32,16 @@ jobs:
         with:
           python-version: ${{ env.min-python-version }}
 
+      - name: Set Date
+        run: echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+        shell: bash
```
Comment (Pwuts) [inline .github/workflows/ci.yml]: bash is the default shell on Ubuntu so this is redundant

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -112,6 +122,16 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: Set Date
+        run: echo "DATE=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+        shell: bash
```
Comment (Pwuts) [inline .github/workflows/ci.yml]: also unnecessary

**Review by Pwuts — CHANGES_REQUESTED** [review]

**Comment by BaseInfinity** [comment]
> @BaseInfinity have the latest packages every day. otherwise, if our requirements.txt doesn't change for 2 weeks, our packages go stale.

Ahh gotcha I wasn't aware we are pulling in the latest versions on a daily basis that makes sense

One potential gotcha I've run into by not locking down specific package versions is getting breaking changes unexpectedly, the flipside is we never updated packages lol so there's that lol

Any ways cool stuff

File: tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py (hunk 1 above)
Comment (waynehamadi) [reply tests/integration/challenges/information_retrieval/test_information_retrieval_challenge_b.py]: yeah here this test uses content not static, there is some metadata that changes apparently. I removed the skip so we can merge this PR and talk about that in a separate PR.

**Review by richbeales — APPROVED** [review]

**Pull request merged**